cmd/agent/app/run.go

```diff
@@ -387,12 +388,15 @@ func StartAgent() error {
 	// Start OTLP intake
 	if otlp.IsEnabled(config.Datadog) {
 		var err error
+		inventories.SetAgentMetadata(inventories.AgentOTLPEnabled, true)
 		common.OTLP, err = otlp.BuildAndStart(common.MainCtx, config.Datadog, demux.Serializer())
 		if err != nil {
 			log.Errorf("Could not start OTLP: %s", err)
 		} else {
 			log.Debug("OTLP pipeline started")
 		}
+	} else {
+		inventories.SetAgentMetadata(inventories.AgentOTLPEnabled, false)
```

Comment:
Consider assigning `otlp.IsEnabled(config.Datadog)` to a variable and calling `SetAgentMetadata` once.